src/test/scala/algolia/integration/RulesIntegrationTest.scala

```diff
@@ -62,6 +62,16 @@ class RulesIntegrationTest extends AlgoliaTest {
       }
     }
 
+    it("should not save a rule with empty objectID") {
+      val f = client.execute {
+        save rule generateRule("") inIndex indexName
+      }
+
+      whenReady(f.failed) { res =>
+        res shouldBe an[AlgoliaClientException]
```

Comment:
just to be sure could you also check the content of the exception? To be sure we are not catching another `AlgoliaClientException`?

Reply:
Good point. Perhaps creating an `AlgoliaMissingObjectIDException` is even better but overkill. What do you think?

Reply:
I agree it might be overkill. I would stick to a generic exception with a good error message.

Reply:
👍